packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py

```diff
@@ -221,20 +223,81 @@ def _get_decision(
+        # here we calculate the bet amount per pool's token
+        bet_per_token = net_bet_amount / BINARY_N_SLOTS
+
+        # calculate the number of the traded tokens
+        prices = bet.outcomeTokenMarginalPrices
+        if prices is None:
+            return 0, 0
+        tokens_traded = [int(bet_per_token / prices[i]) for i in range(BINARY_N_SLOTS)]
+
+        # get the shares for the answer that the service has selected
+        selected_shares = tokens_traded.pop(vote)
+
+        # get the shares for the opposite answer
+        other_shares = tokens_traded.pop()
+
+        # get the number of tokens in the pool for the answer that the service has selected
+        selected_type_tokens_in_pool = token_amounts.pop(vote)
+
+        # get the number of tokens in the pool for the opposite answer
+        other_tokens_in_pool = token_amounts.pop()
+
+        # the OMEN market then trades the opposite tokens to the tokens of the answer that has been selected,
+        # preserving the balance of the pool
+        # here we calculate the number of shares that we get after trading the tokens for the opposite answer
+        tokens_remaining_in_pool = int(k / (other_tokens_in_pool + other_shares))
+        swapped_shares = selected_type_tokens_in_pool - tokens_remaining_in_pool
+
+        # calculate the resulting number of shares if the service would take that position
+        num_shares = selected_shares + swapped_shares
+        # calculate the available number of shares
+        price = prices[vote]
+        available_shares = int(selected_type_tokens_in_pool * price)
+
+        return num_shares, available_shares
+
+    def _is_profitable(self, confidence: float, vote: int) -> bool:
         """Whether the decision is profitable or not."""
+        bet = self.synchronized_data.sampled_bet
+        bet_amount = self.params.get_bet_amount(confidence)
+        net_bet_amount = int(bet_amount * (1 - self.wei_to_native(bet.fee)))
+        num_shares, available_shares = self._calc_binary_shares(net_bet_amount, vote)
+        mech_price = self.synchronized_data.mech_price
         bet_threshold = self.params.bet_threshold
 
-        if bet_threshold < 0:
+        if bet_threshold <= 0:
             self.context.logger.warning(
-                f"A negative bet threshold was given ({bet_threshold}), "
-                f"which means that the profitability check will be bypassed!"
+                f"A non-positive bet threshold was given ({bet_threshold}). The threshold will be disabled, "
+                f"which means that any non-negative potential profit will be considered profitable!"
             )
-            return True
+            bet_threshold = 0
 
-        bet_amount = self.params.get_bet_amount(confidence)
-        fee = self.synchronized_data.sampled_bet.fee
-        return bet_amount - fee >= bet_threshold
+        potential_net_profit = num_shares - net_bet_amount - mech_price - bet_threshold
```

Comment:
I believe that the mech's price should be removed from the profitability calculation as this has already been paid. 
If `0 < num_shares - net_bet_amount - bet_threshold < mech_price`, then the service would at least have a chance to cover a part of the mech's costs plus the threshold's amount.

Reply:
Ok! We need to add the mech cost to the outer decision making problem later

Reply:
Removed on 30bf870d342dd3bb85727569179194dd5d3e2813 ans opened https://github.com/valory-xyz/trader/issues/75.